interface/skimUtils.h

```diff
@@ -644,4 +644,59 @@ pair <TVector2, TVector2> getShiftedMET_mes (TVector2 MET, bigTree & theBigTree,
   return make_pair(shiftedMET_muup, shiftedMET_mudown);
 }
 
+pair <TVector2, TVector2> getShiftedMET_electrons(TVector2 met, bigTree &theBigTree, int pairType, bool isScale)
+{
+  // do nothing in the tautau channel (electrons faking taus)
+  // it is already taken into account in the getShiftedMET_tes_ees() function
+  if(pairType == 2) {
+	return make_pair(met, met);
+  }
+  
+  TVector2 met_up, met_do;
+  double metx_up = met.Px(), metx_do = met.Px();
+  double mety_up = met.Py(), mety_do = met.Py();
+
+  TLorentzVector tlv_new_up;
+  TLorentzVector tlv_new_do;
+
+  for (unsigned int idau=0 ; idau<theBigTree.daughters_px->size() ; ++idau)
+	{
+	  TLorentzVector tlv_old(theBigTree.daughters_px->at(idau),
+							 theBigTree.daughters_py->at(idau),
+							 theBigTree.daughters_pz->at(idau),
+							 theBigTree.daughters_e->at(idau));
+
+	  if (abs(theBigTree.genpart_pdg->at(idau))==11) {
```

Comment:
Don't we need to ask for the pdgId of the reconstructed object here instead of the gen match? The corrections are only stored as userFloats for pat::Electrons ... if the daughter here is not a reconstructed electron, but has a genmatch to an electron, then the way I understand this you would use the "dummy" values for energyScaleUp etc. set to -999 [here](https://github.com/LLRCMS/LLRHiggsTauTau/blob/106X_HH_UL/NtupleProducer/plugins/HTauTauNtuplizer.cc#L2982-L2987)

Reply:
From a chat with @kramerto, we decided that the correct thing to do is to use the reconstructed particle type: ```oph.isElectron(theBigTree.particleType->at(idx);)```.

Reply:
I think you still need to change this here